File: plugins/Aspose_Pdf_Java_for_PHP/src/Aspose/Pdf/WorkingWithPages/GetNumberOfPages.php
<?php

namespace Aspose\Pdf\WorkingWithPages;

use com\aspose\pdf\Document as Document;

class GetNumberOfPages {

    public static function run($dataDir=null)
    {
        # Create PDF document
        $pdf = new Document($dataDir . 'input1.pdf');

        $page_count = $pdf->getPages()->size();

        print "Page Count:" . $page_count . PHP_EOL;
    }

}

File: plugins/Aspose_Pdf_Java_for_Ruby/lib/asposepdfjava/Pages/getnumberofpages.rb
module Asposepdfjava
  module GetNumberOfPages
    def run
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Create PDF document
        pdf = Asposepdfjava.java_class('com.aspose.pdf.Document').new(data_dir + 'input1.pdf')
        
        page_count = pdf.getPages().size()

        puts "Page Count:" + page_count.to_s
    end
  end
end
